js/src/components/shipping-rate-section/minimum-order-card/minimum-order-input-control.js

```diff
@@ -41,23 +44,7 @@ const MinimumOrderInputControl = ( props ) => {
 					<MinimumOrderInputControlLabelText
 						countries={ countries }
 					/>
-					<AppButtonModalTrigger
-						button={
-							<Button
-								className="gla-minimum-order-input-control__edit-button"
-								isTertiary
-							>
-								{ __( 'Edit', 'google-listings-and-ads' ) }
-							</Button>
-						}
-						modal={
-							<EditMinimumOrderModal
-								countryOptions={ countryOptions }
-								value={ value }
-								onChange={ handleEditChange }
-							/>
-						}
-					/>
+					{ labelButton }
```

Comment:
Personally, I'd keep the label button implementation here. It will simplify the use in js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.js and make `MinimumOrderInputControl` self-sufficient component with `onDelete` and `onChange`.

Reply:
See https://github.com/woocommerce/google-listings-and-ads/compare/feature/refactor-minimum-order-modal...feature/refactor-minimum-order-modal-labelButton

Reply:
I thought about a better implementation, by having a generic reusable `EditNumberInputControl` that can be used in shipping rate, shipping time and minimum order, so we don't need to have `MinimumOrderInputControl` etc. See issue https://github.com/woocommerce/google-listings-and-ads/issues/1444. 🙂

Reply:
I see.

But, I wouldn't squeeze #1444 here, and rather solve it by a separate PR. As I may argue, that what we have here is the best solution for 1444. For example, at least the `Edit` button could be a part of such `*InputControl`, and only the modal would be provided by a consumer.

After my commit, the component is closer to what it was before, leaving the clean field to address #1444 separately.

Reply:
> But, I wouldn't squeeze #1444 here, and rather solve it by a separate PR.

Yes, I plan to solve it in a separate PR, not in this one. 🙂 

> After my commit, the component is closer to what it was before, leaving the clean field to address https://github.com/woocommerce/google-listings-and-ads/issues/1444 separately.

Yeah, I understand your point. However, I'm also thinking that it is good for this PR now to be similar to the shipping rate side of things, so when I work on #1444 later, because shipping rate and minimum order are similar, it is also somewhat easy to make the code change, and easy for reviewer. 

I think it is fine for us to leave this concern about label button here, because the concern will go away when we address #1444 (hopefully soon 🙂).

Reply:
> I plan to solve it in a separate PR, not in this one. 

But the change is here. So you are partially including the #1444 fix here.

And in my opinion, this change does not add value and does not simplify the minimum order modal, but the opposite - it makes the logic more dislocated and adds more unnecessary code.

> However, I'm also thinking that it is good for this PR now to be similar to the shipping rate side of things, so when I work on #1444 later, because shipping rate and minimum order are similar, it is also somewhat easy to make the code change, and easy for reviewer.
> 
> I think it is fine for us to leave this concern about label button here, because the concern will go away when we address #1444 (hopefully soon ).


Speaking of a future review of #1444, to be the situation look like, we are here introducing more bloat and more concenrns in the name of "refactoring" to be refactored in the future. It does not sound like a courtesy for the future reviewer, but adding more work there and maintaining questionable implementation in the meantime.

As a #1444 reviewer, I'd rather see the older (pre-shipping rate https://github.com/woocommerce/google-listings-and-ads/pull/1424) implementation in the diff, to be able to evaluate that the #1444 fix delivers something better than what we had. And not just something better than post-#1424, but potentially worse than pre-#1424.

For me "leaving the concern of a button" would actually mean not changing the implementation of `labelButton` in this PR. Then approaching #1444 fixing without a bias of already made solution

Reply:
> So you are partially including the #1444 fix here.

Well, yes, it is kind of a partial fix.

> this change does not add value and does not simplify the minimum order modal, but the opposite - _it makes the logic more dislocated and adds more unnecessary code_.
> [...]
> Speaking of a future review of #1444, to be the situation look like, we are here _introducing more bloat and more concenrns_ in the name of "refactoring" to be refactored in the future. It does not sound like a courtesy for the future reviewer, but adding more work there and maintaining questionable implementation in the meantime.

I don't quite agree with the _above emphasized statement_ (maybe I don't see the way you see it?). I think it is a partial fix towards a better approach, in the following ways: 

- We move the modal out of the `MinimumOrderInputControl`, and put the modal in the parent `MinimumOrderCard`.
- Because of the above point, `MinimumOrderInputControl` does not need `countryOptions` and `onDelete` props.

Even if we don't do the above now in this PR, when we work on issue #1444, we shall do the above anyway, replacing `MinimumOrderInputControl` with `EditNumberInputControl` with `onEditClick` prop. So, we are doing "part of #1444" early in this PR. I don't see how the above "introduce more bloat and concerns". 😅 

> As a #1444 reviewer, I'd rather see the older (pre-shipping rate #1424) implementation in the diff, _to be able to evaluate that the #1444 fix delivers something better than what we had. And not just something better than post-#1424, but potentially worse than pre-#1424._
> 
> For me "leaving the concern of a button" would actually mean not changing the implementation of `labelButton` in this PR. Then _approaching #1444 fixing without a bias of already made solution_

Thanks, this is an interesting perspective, and I think this is fair 🙂 👍  I have made the changes in [`193cb19` (#1410)](https://github.com/woocommerce/google-listings-and-ads/pull/1410/commits/193cb19768843bfbfc13b8bba2dca856a8e00c86). 🙏